src/gymFunctions.cpp

```diff
@@ -0,0 +1,169 @@
+#include "../include/kindyn/gymFunctions.h"
+#include <iostream>
+gymFunctions::gymFunctions(cardsflow::kindyn::Robot* robot, int id, bool respect_limits){
+    if (!ros::isInitialized()) {
+        int argc = 0;
+        char **argv = NULL;
+        ros::init(argc, argv, "Gym functions", ros::init_options::NoSigintHandler);
+    }
+    nh = ros::NodeHandlePtr(new ros::NodeHandle);
+    spinner.reset(new ros::AsyncSpinner(0));
+    spinner->start();
+    ROS_INFO("Gym functions");
+
+    training_robot = robot;
+    training_with_limits = respect_limits;
+
+    gym_step = nh->advertiseService("/instance" + to_string(id) + "/gym_step", &gymFunctions::GymStepService,this);
+    gym_read_state = nh->advertiseService("/instance" + to_string(id) + "/gym_read_state", &gymFunctions::GymReadStateService,this);
+    gym_reset = nh->advertiseService("/instance" + to_string(id) + "/gym_reset", &gymFunctions::GymResetService,this);
+    gym_goal = nh->advertiseService("/instance" + to_string(id) + "/gym_goal", &gymFunctions::GymGoalService,this);
+    fmt = Eigen::IOFormat(4, 0, " ", ";\n", "", "", "[", "]");
+}
+
+
+bool gymFunctions::GymStepService(roboy_simulation_msgs::GymStep::Request &req,
+                                  roboy_simulation_msgs::GymStep::Response &res){
+    training_robot->update();
+
+    Map<VectorXd> action(req.set_points.data()  , training_robot->number_of_cables);
+    training_robot->Ld[0]= action;  //Commanding cable velocity for simulation
+    //training_robot->l[i] = action;     //Commanding cable length for hardware
+
+    if(!training_robot->getExternalRobotState())
+        training_robot->forwardKinematics(req.step_size);
+
+    ROS_INFO_STREAM_THROTTLE(5, "Ld = " << training_robot->Ld[0].format(fmt));
+
+    training_robot->write();
+    if(training_with_limits){
+        res.feasible = isFeasible(training_robot->getLimitVector(0), training_robot->getLimitVector(1), training_robot->q[0], training_robot->q[1]);
```

Comment:
isFeasible should be independent of the number of dimensions (in this case 2).
What about `isInside(VectorXd point, List<VectorXd> bounds);`
Maybe there is a data structure that after initializing we can use like `if bounds.contains(VectorXd point)`